### PR TITLE
#161924003 Add endpoint to update a user's role

### PR DESCRIPTION
**Pull request by PriscillaSam**
#### What does this PR do?
Create endpoint to update a user's role. **PUT** `/api/v1/admin/users/roles`.

#### Description of Task to be completed
- add controller function for the endpoint
- test function and route with mocha and chai
- add documentation for the endpoint

#### How should this be manually tested?
- on the terminal run `git clone https://github.com/andela/haven-ah-backend.git`.
- run `git checkout ft-update-user-role-161924003`.
- ensure that PostgreSQL is running on your local machine.
- set up databases (test and development) following the `.env.sample` file.
- run `npm install` to install the npm packages.
- run `npm test` to run the unit and integration tests.
To test on postman

- run `npm run start-dev` and open up postman.

Authentication is required to perform this operation and is only available to the `SuperAdmin` role.

#### Screenshots where applicable

_Response when wrong enum type is passed_

screen shot 2018-11-14 at 2 05 47 pm

_Response when user with username supplied does not exist._

screen shot 2018-11-14 at 2 06 35 pm

 _Response when the operation is successful_

screen shot 2018-11-14 at 2 09 31 pm

#### Any background context to provide
The create user function was modified to include a role parameter.

#### What are the relevant pivotal tracker stories?
 #161924003

### CODE REVIEW

**Comment by victor-nweze-deel** [comment]
Looks good

**Review by seyi-adeleke — CHANGES_REQUESTED** [review]
there's a diff in your package lock

File: server/controllers/user.js [hunk 1]
```diff
@@ -481,6 +481,59 @@ class User {
...
+    }, username);
+    const { id, firstName, lastName } = updated;
+
+    const updatedUser = {
+      id,
+      name: `${firstName} ${lastName}`,
+      role: updated.role
+    };
+
+    return goodHttpResponse(
```
Comment (TheoOkafor) [inline server/controllers/user.js:530]: Good stuff. @PriscillaSam 
Nice work

**Review by TheoOkafor — COMMENTED** [review]
It looks good. 👍

**Review by seyi-adeleke — APPROVED** [review]

**Pull request merged**